sql/updates/world/4.3.4/2021_01_21_01_world.sql

```diff
@@ -0,0 +1,1452 @@
+(50574,107,5855,0),
+(56796,107,657,0);
+
+-- add missing creature_trainer for Cooking Trainers
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (1355,1699,3067,3087,3399,4210,4552,4894,5159,6286,8306,16253,16277,16676,16719,17246,18987,18988,18993,19185,19369,26905,26953,26972,26989,28705,33587,34785,54232);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES
+(1355,136,5853,0),
+(1699,136,5853,0),
+(3067,136,5854,0),
+(3087,136,5853,0),
+(3399,136,4747,0),
+(4210,136,5853,0),
+(4552,136,5854,0),
+(4894,136,12847,0),
+(5159,136,4748,0),
+(6286,136,2021,0),
+(8306,136,5854,0),
+(16253,136,7171,0),
+(16277,136,5854,0),
+(16676,136,5854,0),
+(16719,136,5853,0),
+(17246,136,7388,0),
+(18987,136,7818,0),
+(18988,136,7818,0),
+(18993,136,7879,0),
+(19185,136,8460,0),
+(19369,136,0,0),
+(26905,136,9985,0),
+(26953,136,9987,0),
+(26972,136,9988,0),
+(26989,136,9986,0),
+(28705,136,9772,0),
+(33587,136,0,0),
+(34785,136,0,0),
+(54232,136,12838,0);
+
+-- add missing creature_trainer for Fishing Trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (1651,1680,1683,1700,2834,3028,3179,3607,4156,4573,5161,5690,5938,5941,12032,12961,14740,16774,16780,17101,18018,18911,23896,26909,26957,26993,28742,44975,50570,56068);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES
+(1651,10,5665,0),
+(1680,10,5665,0),
+(1683,10,5665,0),
+(1700,10,5665,0),
+(2834,10,5665,0),
+(3028,10,5665,0),
+(3179,10,5665,0),
+(3607,10,5665,0),
+(4156,10,5665,0),
+(4573,10,12887,0),
+(5161,10,4742,0),
+(5690,10,5665,0),
+(5938,10,5665,0),
+(5941,10,5665,0),
+(12032,10,5665,0),
+(12961,10,5665,0),
+(14740,10,6087,1),
+(16774,10,5665,0),
+(16780,10,5665,0),
+(17101,10,7361,0),
+(18018,10,7609,1),
+(18911,10,10437,0),
+(23896,10,8826,0),
+(26909,10,0,0),
+(26957,10,0,0),
+(26993,10,0,0),
+(28742,10,9833,0),
+(44975,10,12443,0),
+(50570,10,5665,0),
+(56068,10,5665,0);
+
+-- add missing creature_trainer for Archaeology Trainers.
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (39718,47346,47382,47569,47571,47572,47575,47579,51997,53421);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES
+(39718,373,11821,0),
+(47346,373,11821,0),
+(47382,373,11821,0),
+(47569,373,11821,0),
+(47571,373,12410,0),
+(47572,373,11821,0),
+(47575,373,11821,0),
+(47579,373,11821,0),
+(51997,373,12725,0),
+(53421,373,12850,0);
+
+-- add missing creature_trainer for Riding Trainers
+DELETE FROM `creature_trainer` WHERE `CreatureId` IN (16280);
+INSERT INTO `creature_trainer` (`CreatureId`, `TrainerId`, `MenuId`, `OptionIndex`) VALUES
+(16280,46,8275,0);
+
+-- correct existing creature_trainer
+UPDATE `creature_trainer` SET `MenuId` = 10138 WHERE `CreatureId` = 28699;
+
+-- change a cooking trainer's gossip menu to match the creature_trainer record.
+UPDATE `creature_template` SET `gossip_menu_id` = 10569 WHERE `entry` = 34786;
+
+-- correct trainer npcFlags
+-- herbalism
```

Comment:
if you want to update the npcsflags, please use   SET  `npcflag`= `npcflag` | 0x10 | 0x40   so you add the trainer and profession trainer flag to existing flags only instead of overwriting the present field entirely